codec-sockjs/src/main/java/io/netty/handler/codec/sockjs/transport/RawWebSocketTransport.java

```diff
@@ -0,0 +1,240 @@
+        if (msg instanceof FullHttpRequest) {
+            handleHttpRequest(ctx, (FullHttpRequest) msg);
+        } else if (msg instanceof WebSocketFrame) {
+            handleWebSocketFrame(ctx, (WebSocketFrame) msg);
+        } else {
+            ctx.fireChannelRead(msg);
+        }
+    }
+
+    private static boolean checkRequestHeaders(final ChannelHandlerContext ctx, final HttpRequest req) {
+        if (req.method() != GET) {
+            ctx.writeAndFlush(responseFor(req)
+                    .methodNotAllowed()
+                    .header(CONTENT_LENGTH, 0)
+                    .allow(GET)
+                    .buildResponse())
+            .addListener(ChannelFutureListener.CLOSE);
+            return false;
+        }
+
+        final String upgradeHeader = req.headers().get(UPGRADE);
+        if (upgradeHeader == null || !"websocket".equals(upgradeHeader.toLowerCase())) {
+            ctx.writeAndFlush(responseFor(req)
+                    .badRequest()
+                    .content("Can \"Upgrade\" only to \"WebSocket\".")
+                    .contentType(HttpResponseBuilder.CONTENT_TYPE_PLAIN)
+                    .buildFullResponse()).addListener(ChannelFutureListener.CLOSE);
+            return false;
+        }
+
+        String connectHeader = req.headers().get(CONNECTION);
+        if (connectHeader != null && "keep-alive".equals(connectHeader.toLowerCase())) {
+            req.headers().set(CONNECTION, HttpHeaderValues.UPGRADE);
+            connectHeader = HttpHeaderValues.UPGRADE.toString();
+        }
+        if (connectHeader == null || !"upgrade".equals(connectHeader.toLowerCase())) {
+            ctx.writeAndFlush(responseFor(req)
+                    .badRequest()
+                    .content("\"Connection\" must be \"Upgrade\".")
+                    .contentType(HttpResponseBuilder.CONTENT_TYPE_PLAIN)
+                    .buildFullResponse()).addListener(ChannelFutureListener.CLOSE);
+            return false;
+        }
+        return true;
+    }
+
+    private void handleHttpRequest(final ChannelHandlerContext ctx, FullHttpRequest req) {
+        try {
+            if (!checkRequestHeaders(ctx, req)) {
+                return;
+            }
+            ctx.attr(REQUEST_KEY).set(req);
+            final String wsUrl = webSocketLocation(config.isTls(), req, TransportType.WEBSOCKET.path());
+            final WebSocketServerHandshakerFactory wsFactory = new WebSocketServerHandshakerFactory(wsUrl,
+                    protocols, false);
+            handshaker = wsFactory.newHandshaker(req);
+            if (handshaker == null) {
+                WebSocketServerHandshakerFactory.sendUnsupportedVersionResponse(ctx.channel());
+            } else {
+                final ChannelFuture handshakeFuture = handshaker.handshake(ctx.channel(), req);
+                handshakeFuture.addListener(new ChannelFutureListener() {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Reduce this anonymous class number of lines from 26 to at most 20, or make it a named class. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS1188)